keanu-project/src/main/java/io/improbable/keanu/distributions/continuous/MultivariateGaussian.java

```diff
@@ -35,10 +35,10 @@ public DoubleTensor sample(long[] shape, KeanuRandom random) {
 
     @Override
     public DoubleTensor logProb(DoubleTensor x) {
-        final double dimensions = numberOfDimensions();
+        final long dimensions = numberOfDimensions();
         final double kLog2Pi = dimensions * LOG_2_PI;
         final double logCovDet = Math.log(covariance.determinant());
-        DoubleTensor xMinusMu = x.minus(mu);
+        DoubleTensor xMinusMu = x.minus(mu).reshape(dimensions, 1);
         DoubleTensor xMinusMuT = xMinusMu.transpose();
```

Comment:
Given we know that we know that mu is a 3x1 can't we just reshape it to a 1x3 to do the transpose instead?

Reply:
Good idea.